### ISSUE TITLE
Polybar kills all children of processes it spawns?

### ISSUE THREAD

**Issue by apetresc** [jaagr/polybar#770]
So, I have a shell script that launches `mpv` in the background to connect to my remote MPD server:

```bash
$ cat $(which stream-mpd)
#!/bin/bash
pkill -f "mpv http://mysever"
mpv http://myserver &
```
If I launch this from the command line, everything works as it should, `mpv` runs in the background even if I close the terminal, etc. However, if I try to hook it up to my `mpd` module in polybar (so that I can actually listen to the stream that I am remote-controlling with the module) like so:

```
format-online = %{A3:stream-mpd:}<label-song> %{A}  <icon-prev> <icon-seekb> <icon-stop> <toggle> <icon-seekf> <icon-next>  <icon-repeat> <icon-random>
```
then something strange happens: the `mpv` process is killed as soon as `stream-mpd` exits. I can confirm this by adding a `sleep 10` as the last line of my script: polybar will freeze up for 10 seconds since the script is running, but then once the 10 seconds are up, `mpv` is killed too! I can't explain why that happens - is polybar going out of its way to clean up the children of all the processes it spawns? Is there a way to accomplish what I'm after (which is to start a background task as an input trigger)?

**Comment by patrick96** [comment]
I don't exactly know how polybar handles its child processes, but it makes sense that it cleans up after itself, otherwise you end up with a bunch of processes spawned by polybar which run in the background (not necessarily doing anything or outputting text to the bar without the bar listening for it) or you may even get zombie processes, if they're not handled correctly.

But what you want to achieve is possible inside the script, with one of the following ways:

**nohup:**
```bash
nohup mpv http://myserver &>/dev/null &
```

**disown:**
```bash
mpv http://myserver &
disown
```

**setsid:**
```bash
setsid mpv http://myserver &>/dev/null < /dev/null &
```

The `setsid` version is probably the one that will always work, it directly attaches the process to the init process. It is possible that if you use the `nohup` or `disown` variants that the mpv process is still terminated when the script is terminated but that may depend on the shell and on how the script is terminated.

**Comment by apetresc** [comment]
Thank you for the super-thorough explanation. The `setsid` solution is indeed best. Cheers!

**Comment by patrick96** [comment]
Just FYI, I have just pushed #2248 which will make polybar no longer kill children spawned through click (and scroll) commands. Programs launched through polybar now act more or less the same as if they were started from some application launcher.

**Comment by pawel-szopinski** [comment]
@patrick96 I am on 3.5.0, and now, even the simplest command executed with mouse click (e.g. notify-send "hello" "world") leaves a zombie process. See the screenshot below. These processes do not seem to disappear after a while. They (obviously) are removed after polybar is restarted.

![image](https://user-images.githubusercontent.com/25465830/101370974-7f85c100-38aa-11eb-925d-e9f4d76883db.png)

**Comment by patrick96** [comment]
@pawel-szopinski Hmm, I though I tested this exact thing, but I guess I did not. I fixed this in #2285, this will be in the 3.5.1 release.

**Comment by pawel-szopinski** [comment]
@patrick96 big thanks for the quick fix! I can confirm it is indeed fixed in 3.5.1.

**Comment by apetresc** [comment]
Thank you!
